source/Engine.cpp

```diff
@@ -1623,6 +1623,31 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 		// self-destruct.
 		if(ship->IsDestroyed())
 		{
+			if(!ship->CoreShip().empty())
+			{
+				const Ship *shipToPlace = GameData::Ships().Get(ship->CoreShip());
+				if(shipToPlace->IsValid())
+				{
+					const shared_ptr<Ship> escapeShip = make_shared<Ship>(*shipToPlace);
+					escapeShip->Recharge();
+					escapeShip->SetName(ship->Name());
+					escapeShip->SetGovernment(ship->GetGovernment());
+					if(ship->CorePersonality().IsValid())
+						escapeShip->SetPersonality(ship->CorePersonality());
+					else
+						escapeShip->SetPersonality(ship->GetPersonality());
```

Comment:
You can do this in a trinary operation:
corepers = ship-CorePersonality();
escapeShip->SetPersonality(corepers-IsDefined() ? ... : ...)

Reply:
if you say so